QPM/Providers/RemoteQPMDependencyResolver.cs

```diff
@@ -169,22 +169,35 @@ private void CopyAdditionalData(JsonElement elem, string root, string dst)
             // TODO: Hash check additional data
             foreach (var item in elem.EnumerateArray())
             {
-                var location = Path.Combine(root, item.GetString());
+                var location = Path.Combine(root, item.GetString()!);
                 if (File.Exists(location))
                 {
-                    var dest = Path.GetFullPath(Path.Combine(dst, item.GetString()));
+                    var dest = Path.GetFullPath(Path.Combine(dst, item.GetString()!));
+
+                    Utils.CreateDirectory(Path.GetDirectoryName(dest)!);
 
                     Utils.SymlinkOrCopyFile(location, dest);
                 }
                 else if (Directory.Exists(location))
                 {
-                    var dest = Path.GetFullPath(Path.Combine(dst, item.GetString()));
-                    if (Directory.Exists(dest))
-                        Utils.DeleteDirectory(dest);
 
-                    // Create parent directories
-                    Utils.CreateDirectory(dst);
+                    var dest = Path.GetFullPath(Path.Combine(dst, item.GetString()!));
+                    // Add an extension so it "thinks" it's a file to get the parent directory easily
+                    var destFullParent =
+                        Path.GetDirectoryName(Path.TrimEndingDirectorySeparator(dest) + ".s")!;
```

Comment:
Uhh, why?

Just do .TrimEnd(directory separator char) and call GetDirectoryName

Reply:
Are you sure that won't just return the directory itself? I don't think it goes up one level, though I may be wrong

Reply:
As long as it doesn't end with a `/`, it goes up a level. Here's a link if you don't believe me: [https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getdirectoryname?view=net-5.0](yoink)

Reply:
oh I'm stupid, alright I'll do that